server/filestore.go

```diff
@@ -4198,7 +4235,7 @@ func (fs *fileStore) PurgeEx(subject string, sequence, keep uint64) (purged uint
 			mb.loadMsgsWithLock()
 			shouldExpire = true
 		}
-		if sequence > 0 && sequence <= l {
+		if sequence > 1 && sequence <= l {
```

Comment:
Trying to replicate logic from above with FWC semantics, which is what I was trying to use in a test and what you did as well. Will write a test for this one.

Reply:
The first command you did should purge if all messages are "foo", if you would have said PurgeEx(">", 1, 0) it would purge, but would not have with a subject.

Reply:
Yes, I know because ">" is handled at the top. This is why I specifically pick a subject to trigger this code.

Reply:
It supposed to do the same as ">" or _EMPTY_ if all subjects match, meaning it should have purged. Added test but I believe behavior is correct now and was not before.